localstack/services/apigateway/provider.py

```diff
@@ -89,6 +93,16 @@ def on_after_init(self):
     def _get_moto_backend(context: RequestContext) -> apigw_models.APIGatewayBackend:
         return apigw_models.apigateway_backends[context.account_id][context.region]
 
+    @staticmethod
+    def _remove_rest_api(context: RequestContext, rest_api_id: str) -> None:
+        store = get_apigateway_store(account_id=context.account_id, region=context.region)
```

Comment:
it removes from Localstack store, should we check on Moto's as well ?

Reply:
The `call_moto` should take care of that as the rest API is contained there, I'm not sure how this could happen in moto